samples/bluetooth/throughput/src/main.c

```diff
@@ -549,17 +567,21 @@ int test_run(const struct shell *shell,
 	/* get cycle stamp */
 	stamp = k_uptime_get_32();
 
-	while (prog < IMG_SIZE) {
+	while (*img_ptr) {
 		err = bt_throughput_write(&throughput, dummy, 244);
 		if (err) {
 			shell_error(shell, "GATT write failed (err %d)", err);
 			break;
 		}
 
 		/* print graphics */
-		printk("%c", img[prog / IMG_X][prog % IMG_X]);
+		str_len = (*img_ptr == '\x1b') ? 6 : 1;
```

Comment:
Yes, because we want to print one character in one step. Escape code is not visible, so I am printing escape code (5 bytes) + first character after it (1 byte) = 6 bytes total.

There are different escape codes that may have different length, but they are not used in this ascii-art. Preparing for every possible escape sequence will be an overkill in this place. I can add comment to `img.file` that only 5-byte escape codes are allowed.